Automattic/wp-calypso
# Stats: Only show tiers higher than site usage

Related: 
- https://github.com/Automattic/wp-calypso/issues/85246
- p1712791967808509/1712791561.443719-slack-C82FZ5T4G

## Proposed Changes

* Existing users will see the tier that matches their site traffic to start with
* Always show six tiers to users, not only when we run out of tiers
* Removed some obsolete code

## Testing Instructions

* Open Calypso Live for a site with 10k+ views `/stats/purchase/:siteSlug`
* Ensure the tier starts from `100k` or higher - you could try jetpack.com
* Open Calypso Live for a site with <10k views `/stats/purchase/:siteSlug`
* Open WordPress.com for the same site
* Ensure the tiers and prices match
* Purchase `100k` tier
* Open Calypso Live for the site again `/stats/purchase/:siteSlug`
* Ensure you see six tiers with correct prices

<img width="655" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/96799256-d78f-45ec-8b5b-418c8f26cba8">


## Pre-merge Checklist
